trace/backend_collector.go

```diff
@@ -435,7 +437,13 @@ func (c *BackendCollector) flushPairCache(cutoffTime time.Time) {
 
 			c.queueUpload(e)
 			c.pairCache.Delete(k)
+
+			if !e.witnessFlushed {
+				bufferTimeSum += now.Sub(e.observationTime)
+			}
 		}
 		return true
 	})
+	bufferLength := float64(bufferTimeSum.Nanoseconds()) / float64(pairCacheCleanupInterval.Nanoseconds())
+	printer.Debugf("Approximate unpaired-witness-cache buffer length: %v", bufferLength)
```

Comment:
I'm not quite sure what this is measuring.

This looks like "how long was this request sitting in the agent" but only for ones that we expire?  It seems like queueUpload() would be a better place to get both paired and unpaired requests?

Reply:
I guess it says "unpaired" right in the log entry.  :)